### PR TITLE
chore(*) prepare 3.2.3 patch release

### PR DESCRIPTION
**Pull request by rainest**
**What this PR does / why we need it**:

3.2.3 for cross-namespace plugin ref fix. Also includes an internal perf fix cause why not.

**Special notes for your reviewer**:

Release issue https://github.com/Kong/kubernetes-ingress-controller/issues/6344

https://github.com/Kong/docs.konghq.com/pull/7578 for the associated doc FWIW.

Changelog to main with backport at https://github.com/Kong/kubernetes-ingress-controller/pull/6345

### CODE REVIEW

**Comment by pmalek** [comment]
Any reason to not cherry pick each commit 1 by 1 and creating separate PRs for easier history inspection in the future?

**Comment by rainest** [comment]
> Any reason to not cherry pick each commit 1 by 1 and creating separate PRs for easier history inspection in the future?

Extra reviews and maybe merge conflicts depending on the order (not in this case though).

What workflow/history review would need them separate? As long as we rebase merge the commits remain separate, so AFAIK all we lose is the title indication we'd get if we squashed and retitled to include the PR. Origin PR is still available in the click-through view (e.g. https://github.com/Kong/kubernetes-ingress-controller/commit/641c73c3897db358ac6c0693003b985e2d4e3eed has no title, but still lists it in the Github metadata), which IMO is good enough for something we probably don't check that often.

**Comment by pmalek** [comment]
> Extra reviews and maybe merge conflicts depending on the order (not in this case though).

I see extra reviews as a good thing rather than a problem since there's more opportunities to spot a potential issue ( even though such a thing shouldn't happen with simple cherry picking).

Conflicts shouldn't arise either if the patches are applied in chronological order.

> What workflow/history review would need them separate? As long as we rebase merge the commits remain separate, so AFAIK all we lose is the title indication we'd get if we squashed and retitled to include the PR. Origin PR is still available in the click-through view (e.g. [641c73c](https://github.com/Kong/kubernetes-ingress-controller/commit/641c73c3897db358ac6c0693003b985e2d4e3eed) has no title, but still lists it in the Github metadata), which IMO is good enough for something we probably don't check that often.

GitHub's UI to review each commit from a PR somewhat works but comments from each of them is bound in a weird way which makes tracing back the commented commit hard.

---

I do see this as a shortcut which can save some review back and forth but at the same time it can make error slipping through the review cracks easier.

File: CHANGELOG.md [hunk 1]
```diff
@@ -91,6 +93,22 @@ Adding a new version? You'll need three changes:
...
+  [#6286](https://github.com/Kong/kubernetes-ingress-controller/pull/6286)
+
+### Fixed
+
+- Services using `Secret`s containing the same certificate as client certificates
+  by annotation `konghq.com/client-cert` can be correctly translated.
+  [#6228](https://github.com/Kong/kubernetes-ingress-controller/pull/6228)
+- Generate one entity for each attached foreign entity if a `KongCustomEntity`
+  resource is attached to multiple foreign Kong entities.
+  [#6280](https://github.com/Kong/kubernetes-ingress-controller/pull/6280)
```
Comment (pmalek) [inline CHANGELOG.md:110]: Since we maintain the changelog in `main`, please cherry pick those changes in changelog after 3.2.3 is released.

**Review by pmalek — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (pmalek) [reply CHANGELOG.md:110]: I can see this is already done in https://github.com/Kong/kubernetes-ingress-controller/pull/6345 👍

**Pull request merged**